logging-6.x.md

```diff
@@ -0,0 +1,176 @@
+# Gravity Logging (for Gravity 6.0 and later)
+
+## Prerequisites
+
+Docker 101, Kubernetes 101, Gravity 101.
+
+## Introduction
+
+Gravity clusters come preconfigured with the logging infrastructure that collects the logs from all running containers, forwards them to a single destination and makes them available for viewing and querying via an API.
+
+In Gravity 6.0 and later the logging stack is based on [Logrange](https://www.logrange.io/) - an open-source streaming database for aggregating application logs and other machine-generated data from multiple sources.
+
+## Pods / Containers Logs Locations
+
+Before diving into Gravity’s logging infrastructure, let’s explore how it is set up in Kubernetes in general.
+
+Kubernetes sets up symlinks in well-known locations for logs of all containers running in the cluster and groups them up in two directories on each node, by pod and by container. The directories where these logs go are `/var/log/pods` and `/var/log/containers` respectively.
+
+Note that these directories reside inside the planet container, not on the host.
+
+On each node, the logs of all containers running in the same pod will be grouped in that pod directory under `/var/log/pods`:
+
+```bash
+planet$ ls -l /var/log/pods/<pod-id>/
+```
+
+If we look at the logs of a particular container, we’ll see that these are in fact symlinks to the actual log files Docker keeps inside its data directory:
+
+```bash
+planet$ ls -l /var/log/pods/01146b3e-3709-11ea-b8d5-080027f6e425/init/
+total 4
+lrwxrwxrwx 1 root root 161 Jan 14 20:04 0.log -> /ext/docker/containers/236f...-json.log
+```
+
+In addition to `/var/log/pods`, Kubernetes also sets up a `/var/log/containers` directory which has a flat structure and the logs of all containers running on the node. The log files are also symlinks that point to the respective files in `/var/log/pods`:
+
+```bash
+planet$ ls -l /var/log/containers/
+total 180
+lrwxrwxrwx 1 root root 66 Jan 15 00:24 bandwagon-6c4b...-lqbll_kube-system_bandwagon-2641....log -> /var/log/pods/0b8e.../bandwagon/1.log
+```
+
+## Logrange Components
+
+Logrange consists of the following main components: collector, aggregator and forwarder. These components can be seen in the following diagram.
```

Comment:
Does it make sense to link to diagram? (adding source) - I may have missed it

Reply:
Oh lol. I forgot to actually include it.